app/services/agreements/iaa_gtc_seeder.rb

```diff
@@ -0,0 +1,28 @@
+module Agreements
+  class IaaGtcSeeder < BaseSeeder
+    private
+
+    def record_class
+      IaaGtc
+    end
+
+    def filename
+      'iaa_gtcs.yml'
+    end
+
+    def primary_attribute_bundle(config)
+      { 'gtc_number' => config['gtc_number'] }
+    end
+
+    def process_config(gtc_number, config)
+      config['partner_account'] =
+        PartnerAccount.find_by!(requesting_agency: config['partner_account'])
+      config['iaa_status'] =
+        IaaStatus.find_by!(name: config['iaa_status'])
+
+      permitted_attrs =
+        %w[gtc_number mod_number start_date end_date estimated_amount iaa_status partner_account]
```

Comment:
In retrospect the attribute used as the key doesn't need to be included here (e.g. `gtc_number`), I'll clean it up in this and the other service objects.